### PR TITLE
implement native realm

### PR DESCRIPTION
**Pull request by danvixent**

### CODE REVIEW

**Review by Subomi — CHANGES_REQUESTED** [review]
Early feedback: Please let's have consistent file names: `file-realm.go` should be `file_realm.go`

Please let's fix this for all affected files. @danvixent

**Comment by danvixent** [comment]
@Subomi done

File: auth/realm/native/native_realm.go [hunk 1]
```diff
@@ -0,0 +1,57 @@
+package native
```
Comment (Subomi) [inline auth/realm/native/native_realm.go:1]: @danvixent This needs to be tested.

**Review by Subomi — CHANGES_REQUESTED** [review]
@danvixent FYI

File: api_key.go [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+	ID        primitive.ObjectID `json:"-" bson:"_id"`
+	UID       string             `json:"uid" bson:"uid"`
+	Role      auth.Role          `json:"role" bson:"role"`
+	Hash      string             `json:"-" bson:"hash"`
+	Revoked   bool               `json:"revoked" bson:"revoked"`
+	ExpiresAt primitive.DateTime `json:"expires_at,omitempty" bson:"expires_at,omitempty"`
+	CreatedAt primitive.DateTime `json:"created_at,omitempty" bson:"created_at"`
+}
+
+type APIKeyRepo interface {
```
[the remark this answers is not in the document]
Comment (danvixent) [reply api_key.go]: done

File: auth/realm/native/native_realm.go (hunk 1 above)
Comment (danvixent) [reply auth/realm/native/native_realm.go:1]: done

**Pull request closed without merging**